code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+			return
+		else if (!isturf(loc))
+			to_chat(src, "<span class='notice'>You need more room to moult.</span>")
+			return
+		else if (stat==UNCONSCIOUS)
+			to_chat(src, "<span class='notice'>You must be awake to moult.</span>")
+			return
+		else if (busy)
+			to_chat(src, "<span class='notice'>You are already doing something.</span>")
+			return
+		else
+			start_moult()
+
+	else
+		to_chat(src, "<span class='notice'>You are already fully mature.</span>")
+
+/mob/living/simple_animal/hostile/grue/proc/start_moult()
+	if(stat==CONSCIOUS&&shadowpower>=moultcost&&!ismoulting&&lifestage<3)
+		shadowpower-=moultcost
+		lifestage++
+		to_chat(src, "<span class='notice'>You begin moulting.</span>")
+		visible_message("<span class='warning'>\The [src] morphs into a chrysalis...</span>")
+		stat=UNCONSCIOUS //go unconscious while moulting
+		ismoulting=1
+		moulttimer=moulttime//reset moulting timer
+		plane = MOB_PLANE //In case grue moulted while hiding
+		var/tempHealth=health/maxHealth //to scale health level
+		if(lifestage==2)
+			desc = "A small grue chrysalis."
+			name = "grue chrysalis"
+			icon_state = "moult1"
+			icon_living = "moult1"
+			icon_dead = "moult1"
+			maxHealth=25 //vulnerable while moulting
+		else if(lifestage==3)
+			desc = "A grue chrysalis."
+			name = "grue chrysalis"
+			icon_state = "moult2"
+			icon_living = "moult2"
+			icon_dead = "moult2"
+			maxHealth=50 //vulnerable while moulting
+		health=tempHealth*maxHealth //keep same health percentage
+	else
+		return
+
+/mob/living/simple_animal/hostile/grue/proc/complete_moult()
+	if(ismoulting&&stat!=DEAD)
+		var/tempHealth=health/maxHealth //to scale health level
+		lifestage_updates()
+		health=tempHealth*maxHealth //keep same health percent
+		stat=CONSCIOUS //wake up
+		ismoulting=0 //is no longer moulting
+		visible_message("<span class='warning'>The chrysalis shifts as it morphs into a grue!</span>")
+		if(lifestage==2)
+			to_chat(src, "<span class='warning'>You finish moulting! You are now a juvenile, and are strong enough to force open doors./span>")
+		else if(lifestage==3)
+			to_chat(src, "<span class='warning'>You finish moulting! You are now fully-grown, and can eat sentient beings to gain their strength.</span>")
+		playsound(src, 'sound/effects/lingextends.ogg', 50, 1)
+	else
+		return
+
+
+/mob/living/simple_animal/hostile/grue/death(gibbed)
+	if(ismoulting)
+		desc="[desc] This one seems dead and lifeless."
+	else
+		playsound(src, 'sound/misc/grue_screech.ogg', 50, 1)
+	..()
+
+/mob/living/simple_animal/hostile/grue/attack_animal(mob/living/simple_animal/M)
+	if(M==src) //Prevent the grue from attacking itself, might help avoid misclicks while attempting to smash lights.
+		return
+	else
+//		if(prob(20)&&lifestage>1)
+//			playsound(src, 'sound/misc/grue_growl.ogg', 50, 1) //occasionally growl while attacking
+		M.unarmed_attack_mob(src)
+
+
+//Reproduction via egglaying.
+/mob/living/simple_animal/hostile/grue/verb/reproduce()
```

Comment:
Ideally you would want this to be a spell, but this is going the extra mile.

Reply:
I'll look into the spells. It could also help hide all the lifestage-specific verbs when they're not usable, if I understand correctly.